src/main/webapp/app/components/Header.tsx

```diff
@@ -40,17 +41,45 @@ export interface IHeaderProps {
 type SubpageLink = {
   title: string;
   link: string;
+  matchedPaths?: PAGE_ROUTE[];
+};
+
+const checkIfNavLinkIsActive = (
+  title: string,
+  location: Location,
+  subPages: SubpageLink[]
+) => {
+  const currentPage: SubpageLink = subPages.find(page => page.title === title)!;
+  const currentLink = currentPage.matchedPaths?.find(link =>
```

Comment:
Don't force using `!` and `?`. `currentPage` and `currentPage.matchedPaths` could be undefined. You just have to handle the scenarios.

Reply:
I see where the issue might lie with the "!". But I assumed a subpage could not turn up as false because a false subpage would be handled elsewhere and return to the homepage. I will fix that though and handle a scenario where there is no subpage. But for the "?" where there is no currentPage.matchedPaths, the expression will return undefined leading the function to return false, so that scenario is already taken care of.

Reply:
I understand it will result in undefined, if you use `!` that bypass the typescript type checks. In this case, it should be SubpageLink|undefined. So conceptually, it makes the code easier to understand.